mirage/seed_image/catalog_seed_image.py

```diff
@@ -2033,91 +1893,37 @@ def parseRADec(self, rastr, decstr):
-            pixelx, pixely = coord_transform.inverse(pixelv2, pixelv3)
-
-        return pixelx, pixely
-
-    def RADecToXY_manual(self, ra, dec):
-        # In this case, the sources are provided as an RA, Dec list,
-        # but no astrometry information is provided. So assume an average
-        # pixel scale and calculate the pixel position of the source from that.
-        # This obviously does not include distortion, and is kind of a last
-        # resort.
-        ra_source = ra * 3600.
-        dec_source = dec * 3600.
-
-        #dist_between, deltaang = self.object_separation([self.ra, self.dec],
-        #                                               [ra_source, dec_source])
-
-        #simple_wcs_obj = {}
-        #simple_wcs_obj["CRPIX1"] = self.refpix_pos["x"]
-        #simple_wcs_obj["CRPIX2"] = self.refpix_pos["y"]
-        #simple_wcs_obj["CRVAL1"] =
-        #simple_wcs_obj["CRVAL2"] =
-        #simple_wcs_obj["CTYPE1"] =
-        #simple_wcs_obj["CTYPE2"] =
-        #simple_wcs_obj["CD1_1"] =
-        #simple_wcs_obj["CD1_2"] =
-        #simple_wcs_obj["CD2_1"] =
-        #simple_wcs_obj["CD2_2"] =
-
-        #wcs_obj = astropy.wcs.WCS(header = simple_wcs_obj)
-        deltara, deltadec = self.object_separation([self.ra, self.dec],
-                                                   [ra_source, dec_source], wcs_obj)
-        # Now translate to deltax and deltay if the
-        # position angle is non-zero
-        #tot_ang = deltaang + (0. - self.params['Telescope']['rotation'] * np.pi / 180.)
+    def RADecToXY_astrometric(self, ra, dec):
+        """Translate backwards, RA, Dec to V2, V3. If a distortion reference file is
+        provided, use that. Otherwise fall back to pysiaf.
 
-        #deltax = dist_between * np.sin(tot_ang) / self.pixscale[0]
-        #deltay = dist_between * np.cos(tot_ang) / self.pixscale[0]
-
-        #pixelx = self.refpix_pos['x'] + deltax
-        #pixely = self.refpix_pos['y'] + deltay
+        Parameters:
+        -----------
+        ra : float
+            Right ascention value, in degrees, to be translated.
 
+        dec : float
+            Declination value, in degrees, to be translated.
 
+        Returns:
+        --------
+        pixelx : float
+            X coordinate value in the aperture corresponding to the input location
 
+        pixely : float
+            Y coordinate value in the aperture corresponding to the input location
+        """
+        loc_v2, loc_v3 = pysiaf.utils.rotations.getv2v3(self.attitude_matrix, ra, dec)
 
+        if self.coord_transform is not None:
+            # Use the distortion reference file to translate from V2, V3 to RA, Dec
+            pixelx, pixely = self.coord_transform.inverse(loc_v2, loc_v3)
+        else:
+            pixelx, pixely = self.siaf.tel_to_sci(loc_v2, loc_v3)
+            # Subtract 1 from SAIF-derived results since SIAF works in a 1-indexed coord system
+            pixelx -= 1
+            pixely -= 1
+        stophere
```

Comment:
This "stophere" line is throwing an error.

Reply:
I did not run into this in any of the tests I made, which seems odd as I had an input point source list with many objects in it.

Reply:
Oops. Sorry. That was leftover from investigating NIRISS issues.